content/kubeone/master/guides/registry_configuration/_index.en.md

```diff
@@ -0,0 +1,134 @@
+  image registry credentials is planned for KubeOne 1.4)
+* if you're using containerd, the image registry must support TLS
+
+If you don't have an image registry, you can check out the
+[Docker Registry][docker-reg-guide] as a possible solution.
+
+## Preloading Images
+
+Another prerequisites for this guide to work is that your image registry has
+all images needed for your cluster to work preloaded.
+
+To make this task easier, we provide the image loader script that:
+
+* pulls all images used by components deployed by KubeOne (CNI,
+  metrics-server...) and Kubeadm (Kubernetes core components and CoreDNS)
+* re-tag those images so the image registry (e.g. `docker.io`) is replaced
+  with the image registry provided by the user
+* push re-tagged images to your (mirror) image registry
+
+The image loader script (`image-loader.sh`) comes in the KubeOne release
+archive, under the `hack` directory. It can also be found on [GitHub in the
+`hack` directory][img-loader]. If you're downloading the script from GitHub,
+it's recommended to switch to the appropriate release branch depending on which
+KubeOne version you're using.
+
+Once you have downloaded the script, you can run it in the following way.
+Make sure to replace `KUBERNETES_VERSION` with the Kubernetes version you plan
+to use (without the `v` prefix), as well as, replace the `TARGET_REGISTRY` with
+the address to your image registry.
+
+```
+KUBERNETES_VERSION=1.22.4 TARGET_REGISTRY=127.0.0.1:5000 ./image-loader.sh
+```
+
+The preloading process can take a several minutes, depending on your
+connection speed.
+
+## Overriding Image Registries
+
+You can override the image registry by adding the `registryConfiguration`
+stanza to your KubeOne configuration file, such as:
+
+```yaml
+apiVersion: kubeone.k8c.io/v1beta2
+kind: KubeOneCluster
+versions:
+  kubernetes: 1.22.4
+cloudProvider:
+  aws: {}
+registryConfiguration:
+  overwriteRegistry: '127.0.0.1:5000'
+  insecureRegistry: false
+```
+
+Make sure to replace the `overwriteRegistry` value with the URL of your image
+registry. If your image registry doesn't support TLS access, make sure to set
+`insecureRegistry` to `true`.
+
+{{% notice warning %}}
+As stated in the prerequisites, if you're using containerd, `insecureRegistry`
+option is not supported, i.e. your image registry **must** support the TLS
+access.
+{{% /notice %}}
+
+With this done, you can reconcile your cluster by running `kubeone apply`.
+
+## Known Issues
+
+TBD (CoreDNS issue)
+
+## Alternatives
+
+{{% notice warning %}}
+We **heavily advise** using the approach described above. This section shows
+possible alternatives that should be used **ONLY** in the case when you are
+**NOT** able to use the approach described above.
+{{% /notice %}}
+
+The alternative to the RegistryConfiguration API is the AssetConfiguration API,
+but it has several caveats:
+
+* the AssetConfiguration API is deprecated and is planned to be removed in
```

Comment:
I think we should restrain from mentioning AssetConfiguration, because it is:
* being deprecated
* is different API

Reply:
I've removed the AssetConfigurtion API from the document.